porch/engine/pkg/engine/engine.go

```diff
@@ -245,7 +246,7 @@ func (cad *cadEngine) UpdatePackageRevision(ctx context.Context, repositoryObj *
 			mutations = append(mutations, mutation)
 
 		default:
-			return nil, fmt.Errorf("updating task of type %q not supported", newTask.Type)
+			return nil, fmt.Errorf("updating task of type %q not supported (oldTask=%s, newTask=%s)", newTask.Type, debug.JSON(oldTask), debug.JSON(newTask))
```

Comment:
I added this logic mostly because I couldn't understand what was going on... turned out to be the ConfigMap nil vs empty problem (immediately below)